### ISSUE TITLE
RSVP flow

### ISSUE THREAD

**Issue by minaorangina** [DRDD2016/api#46]
- [x] database design
- [x] endpoint
- [x] SQL query
### To consider:
* Need to retrieve an invitee's rsvp for an event.  
    * Best to have one big `rsvps` table?  One row == one invitee's rsvp?
    * Each user has personal `rsvp:user_id` table created on signup?
* How to keep track of rsvps for event
    * JSON/arrays of rsvps? (going, not going, maybe)
    * How to update rsvps? If invitee goes from `going` to `not going`, how to delete old entry?

**Comment by minaorangina** [comment]
#### Possible solution: one big `rsvps` table

| event_id  |  user_id | status  |
|---|--|---|
| 1  | 3  | 'going'  |
| 1  | 6  | 'not_going'  | 
| 1  | 9  |  'not_going' | 
| 3 | 11  | 'maybe |

```sql
SELECT status, array_agg(user_id) AS invitees FROM rsvps WHERE event_id = 1;
```
Result:

| status  | invitees |
|---|---|
| going  | {3} |
| not_going  | {6,9} |

**Comment by minaorangina** [comment]
```sql
SELECT user_id, firstname, surname, photo_url FROM users WHERE user_id = 1;

SELECT row_to_json(data) FROM (SELECT user_id, firstname, surname, photo_url FROM users WHERE user_id = 1) data;

SELECT status, (SELECT row_to_json(data) AS invitees FROM (SELECT user_id, firstname, surname, photo_url FROM users WHERE user_id = 1) data) FROM rsvps WHERE event_id = 1 GROUP BY status;


```

**Comment by heron2014** [comment]
Resolved by #49 #51
